fern-aviles/my-design-lite
# Product to group

What's new
-
- Reworked Water object

Set Up
-
- Clone repo
- In your command line:
  - run npm install to install necessary packages
  - run npm run dev to run the project in your web browser

On your browser, go to localhost:5173 and you should see a page with input fields with a black line and text reading "5 feet" under it.

Testing
-
- Click to add a product
- You should be able to resize it and change the angle of rotation using the red controllers
- When deselecting, its controls and info should not appear and when selected they should appear
- Pressure changes water radius size
- Adjusting the arc from either end controllers should change the middle controller
- Adjusting the middle controller should move the end controllers and increase/decrease the radius of the water object
- Ensure correct nozzle is selected on edges of omitted angles
- There should be no issues when changing the angle from 360°
- There is a drag buffer when adding products


Bugs
-
- For MP Rotators: When the radius is between 17.01-18 ft,, maxArc should be 90°. Instead, it's at 210°. (When at 30 PSI, this changes depending on PSI but the behavior is the same)
  - This is due to the MP-2000-90 nozzle at 90° having an extra foot in its radius compared to the other angles in the nozzle. 
  - MP-2000-90 at 180° and 210° have a radius of 17ft
